solr/core/src/java/org/apache/solr/handler/UpdateRequestHandler.java

```diff
@@ -134,12 +135,23 @@ protected void setAssumeContentType(String ct) {
   }
   private Map<String ,ContentStreamLoader> pathVsLoaders = new HashMap<>();
   protected Map<String,ContentStreamLoader> createDefaultLoaders(@SuppressWarnings({"rawtypes"})NamedList args) {
+
     SolrParams p = null;
     if(args!=null) {
       p = args.toSolrParams();
     }
     Map<String,ContentStreamLoader> registry = new HashMap<>();
-    registry.put("application/xml", new XMLLoader().init(p) );
+
+    // Try to load the scripting contrib modules XSLTLoader, and if not, fall back to the XMLLoader.
+    // Enable the XSLTLoader by including the Scripting contrib module.
+    try {
+      final Class<?> clazz = Class.forName( "org.apache.solr.scripting.util.xslt.XSLTLoader" );
```

Comment:
If the scripting contrib is added as a package, Solr core.jar will not be able to load it in this way, I think.

Reply:
This was the part where I didn't like what I did...   And was putting *something* up as a starter..   I really appreciate your and @uschindler input on this.   Definitily lets find another approach!